Adding the correct version of wgrib2 on Orion to convert GRIB2 to netcdf

mrinalbiswas opened the pull request:
## Description of changes
The RT on Orion was failing on the ww3 tasks because it was unable to convert the GRIB2 gfs files to NETCDF. 

Error: 
03/17 09:42:50.410 hafs.ww3init (run.py:349) INFO: Starting: batchexe('/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/impi-2018.4/wgrib2/2.0.8/bin/wgrib2')['/work/noaa/hwrf/noscrub/hafs-input/COMGFSv16/gfs.20200825/12/atmos/gfs.t12z.pgrb2.0p25.f000','-i','-append','-netcdf','[gfs.uvgrd10m.nc](http://gfs.uvgrd10m.nc/)'].in('585:363177507:d=2020082512:UGRD:10 m ...'...,string=True)
netcdf package not installed

The PR will fix the error. 

03/17 09:42:50.434 hafs.ww3init (run.py:453) INFO:   - exit status

## Contributors (optional)
If others worked on this PR besides the author, please include their user names here (using @Mention if possible).

## Tests conducted
What testing has been conducted on the PR thus far? Describe the nature of any scientific or technical tests, including relevant details about the configuration(s) (e.g., cold versus warm start, number of cycles, forecast length, whether data assimilation was performed, etc). What platform(s) were used for testing?
Tested on Orion

## Application-level regression test status
Running the HAFS application-level regression tests is currently performed by code reviewers after the developer creates the initial PR. As regression tests are conducted, the testers should use the checklist below to indicate **successful** regression tests. You may add other tests as needed. If a test fails, do not check the box. Instead, describe the failure in the PR comments, noting the platform where the test failed.

- [ ] Jet
- [ ] Hera
- [x] Orion PASSED
- [ ] WCOSS2

mrinalbiswas committed:
Adding the correct version of wgrib2 on Orion to convert GRIB2 to netcdf

BinLiu-NOAA commented on jobs/JHAFS_WAV_PREP:
```diff
@@ -33,6 +33,11 @@ if [ "${RUN_ENVIR^^}" != "NCO" ]; then
+  # A temporary workaround due to the hpc-stack version of wgrib2 does not work
+  # on Orion to convert grib2 files into netcdf format.
+  if [ ${machine} = orion ]; then
+    export WGRIB2=/apps/wgrib-2.0.8/wgrib2/wgrib2
```
@mrinalbiswas and @BijuThomas-NOAA, will this /apps/wgrib-2.0.8/wgrib2/wgrib2 version works for all other HAFS jobs on Orion, if so, we may change the wgrib2 version in the modulefiles/hafs.orion.lua instead.

mrinalbiswas replied:
@BinLiu-NOAA I can run with this version and let you know. The path on hafs.orion.lua directs wgrib2 to setenv("WGRIB2", "/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/impi-2018.4/wgrib2/2.0.8/bin/wgrib2"). I am not sure whether this is a working version of wgrib2 or not. The default on Orion points to: /apps/wgrib2-3.0.2/wgrib2/wgrib2.

BinLiu-NOAA replied:
@mrinalbiswas, appreciate it if you could help to test if the /apps/wgrib2-3.0.2/wgrib2/wgrib2 version and its corresponding module version can work for the whole HAFS workflow on Orion. In such case, we can just implement this fix inside hafs.orion.lua. Thanks!

KathrynNewman replied:
@mrinalbiswas - I originally approved this PR before Bin's suggestion. I requested a re-review once this has been tested. Thanks!

mrinalbiswas committed:
Merge branch 'hafs-community:production/hafs.v1' into production/hafs.v1

BinLiu-NOAA committed:
Fix a typo bug introduced yesterday when adding some log messages in exhafs_atm_vi.sh.

BinLiu-NOAA committed:
*Update submodule hafs_tracker.fd, which is consistent with
https://github.com/NOAA-GFDL/GFDL-VortexTracker/releases/tag/v1.0.0.
*Corresondingly, the namelist.gettrk_tmp was updated.
From @TimMarchok-NOAA (GFDL), tested in HAFS by @BijuThomas-NOAA.
Note: This new version of GFDL Tracker has the fix for some tracker
failures for some SH storms/cycles.

BinLiu-NOAA committed:
Update submodule hafs_forecast.fd, which contains the updated FV3/upp
with updated HAFS related post control files with the POT variable.
From @LinZhu-NOAA.

BinLiu-NOAA committed:
* Script/workflow cleanups (config, build, cronjob driver, etc.).
  - Default configuration is now HFSA.
  - Add example cronjob drivers: cronjob_hfsa.sh, cronjob_hfsb.sh.
* Add the vi_force_cold_start option to force cold-start the storm vortex when/if desired.
* Move export HDF5_USE_FILE_LOCKING=FALSE into hafs_runcmd.sh.inc.
Notes:
  - Some old config files were moved into parm/tests dir.
  - Some old cronjob drivers were moved into rocoto/examples dir.
  - Only tested on WCOSS2 so far after this round of cleanup.
  - Currently, cronjob_hafs_rt.sh does not work.

BinLiu-NOAA committed:
Cleanup hafs_pre_job.sh.inc and comment out the ulimit line on Orion.

BinLiu-NOAA committed:
Update submodule hafs_utils.fd, which was synced with the its latest
develop branch (removing gotos and using later module versions on WCOSS2).

BinLiu-NOAA committed:
Update submodule hafs_tracker.fd, which reverted the basin ID changes for AA, BB, SI, SP storms.

BinLiu-NOAA committed:
Update submodule hafs_gsi.fd, which has the update to drop off observations when the number of
obs exceeds maxobs. This can avoid the out of bound/dimension issue in GSI. (From @YonghuiWeng-NOAA)

mrinalbiswas committed:
Adding the correct version of wgrib2 for all tasks

mrinalbiswas committed:
Merge branch 'hafs-community:production/hafs.v1' into production/hafs.v1

BinLiu-NOAA committed:
* Add the force_cold_start option, which can overwrite the
  vi_force_cold_start and ww3_force_cold_start options, to force cold
  start atm_vi and WW3 (From @BinLiu-NOAA, @ZhanZhang-NOAA, @lgannoaa).
* Add the vi_min_wind_for_init (default: 9 m/s) to skip atm_vi for weak
  storms when vmax <=vi_min_wind_for_init (From @BinLiu-NOAA, ZhanZhang-NOAA).
* Enable choosing the large HYCOM NHC domain when running South Atlantic (SL/LS) storms.
* Use 'time' instead of 'APRUNS' in cmdfile_ocean in exhafs_emcgraphics.sh (From ChuankaiWang-NOAA).

BinLiu-NOAA committed:
Improve to better handle ww3_force_cold_start in ush/hafs/ww3.py.

mrinalbiswas committed:
Merge branch 'hafs-community:production/hafs.v1' into production/hafs.v1

mrinalbiswas commented:
@BinLiu-NOAA I ran the RT successfully with only the wgrib2 changes in modulefiles/hafs.orion.lua. So we do not need the changes in the jobs/JHAFS_WAVE_PREP.

mrinalbiswas committed:
Revert "Adding the correct version of wgrib2 on Orion to convert GRIB2 to netcdf"

This reverts commit ffd2573b502428593911a75a3d1560a68f01b461.

mrinalbiswas committed:
Revert "Adding the correct version of wgrib2 on Orion to convert GRIB2 to netcdf"

This reverts commit ffd2573b502428593911a75a3d1560a68f01b461.

mrinalbiswas commented:
@BinLiu-NOAA @ZhanZhang-NOAA @KathrynNewman @linlin-pan Please review the PR. RTs passed successfully on Orion

BinLiu-NOAA reviewed (approved):
Looks good to me. Thanks for testing this out and fixing it, @mrinalbiswas!

BinLiu-NOAA commented:
@mrinalbiswas, Could you please help update your PR branch with the latest production/hafs.v1 branch? Meanwhile, if would be beneficial if you can also update the PR description properly. After that, I we can "squash and merge" this PR. Thanks!

mrinalbiswas committed:
Merge branch 'hafs-community:production/hafs.v1' into production/hafs.v1

mrinalbiswas committed:
Adding the correct version of wgrib2 for Orion; This will also eliminate errors related to converting GRIB2 files to NETCDF when running WW3 tasks

mrinalbiswas committed:
Merge branch 'production/hafs.v1' of https://github.com/mrinalbiswas/HAFS into production/hafs.v1

mrinalbiswas committed:
Merge branch 'hafs-community:production/hafs.v1' into production/hafs.v1

mrinalbiswas committed:
Revert "Adding the correct version of wgrib2 for Orion; This will also eliminate errors related to converting GRIB2 files to NETCDF when running WW3 tasks"

This reverts commit 19eeeffa66ebbb996646d2c7e60c359f1335490b.

mrinalbiswas committed:
Merge branch 'hafs-community:production/hafs.v1' into production/hafs.v1

mrinalbiswas commented:
@BinLiu-NOAA Updated my PR branch and updated the PR description. Ready for review.
